### PR TITLE
DPE-2616 Basic profile support

### PR DESCRIPTION
**Pull request by paulomach**
## Issue

Profiles support missing from PG

## Solution

* Created method to build postgresql parameters, based on profile.
* using patroni template to configure parameters
* defaults to `production` profiles (better performance, as done for mysql)
* parameter building is a MVP:
  * `testing` profile set no extra parameters
  * `production` set's minimal needs for IS scenarios (see follow-up)
  * has `available_memory` input, but is not in use: logic is TBD
* integration tests uses `testing` profile

## Folow-up PRs

1. Proper memory detection for parameter calculation
1. React `on-config-change` to allow inflight profile change

### CODE REVIEW

File: src/charm.py [hunk 1]
```diff
@@ -1313,6 +1313,7 @@ def update_config(self, is_creating_backup: bool = False) -> bool:
             backup_id=self.app_peer_data.get("restoring-backup"),
             stanza=self.app_peer_data.get("stanza"),
             restore_stanza=self.app_peer_data.get("restore-stanza"),
+            parameters=self.postgresql.build_postgresql_parameters(self.config["profile"], 0),
```
Comment (marceloneppel) [inline src/charm.py:1316]: Should the user be able to change the profile after the charm was already deployed?

Comment (paulomach) [reply src/charm.py:1316]: Yes - it will need to happen - to keep it simple I will do on follow up PR.

**Review by marceloneppel — APPROVED** [review]
LGTM! Thanks Paulo!

File: lib/charms/postgresql_k8s/v0/postgresql.py [hunk 2]
```diff
@@ -419,3 +419,20 @@ def update_user_password(self, username: str, password: str) -> None:
...
+        Args:
+            profile: the profile to use.
+            available_memory: available memory to use in calculation
+
+        Returns:
+            Dictionary with the PostgreSQL parameters.
+        """
+        logger.debug(f"Building PostgreSQL parameters for {profile=} and {available_memory=}")
+        if profile == "production":
+            return {"shared_buffers": "987MB", "effective_cache_size": "2958MB"}
```
Comment (paulomach) [inline lib/charms/postgresql_k8s/v0/postgresql.py:438]: This static setup reflects the need from IS and will be replaced on next PR

**Review by dragomirp — APPROVED** [review]

**Pull request merged**